source/NVDAObjects/UIA/__init__.py

```diff
@@ -142,127 +224,203 @@ def _getControlFieldForObject(self, obj):
-			pastEnd=tempRange.CompareEndpoints(UIAHandler.TextPatternRangeEndpoint_End,rangeObj,UIAHandler.TextPatternRangeEndpoint_End)>0
-			if pastEnd:
-				tempRange.MoveEndpointByRange(UIAHandler.TextPatternRangeEndpoint_End,rangeObj,UIAHandler.TextPatternRangeEndpoint_End)
-			yield tempRange.clone()
-			if pastEnd:
-				return
-			tempRange.MoveEndpointByRange(UIAHandler.TextPatternRangeEndpoint_Start,tempRange,UIAHandler.TextPatternRangeEndpoint_End)
-		# Ensure that we always reach the end of the outer range, even if the units seem to stop somewhere inside
-		if tempRange.CompareEndpoints(UIAHandler.TextPatternRangeEndpoint_End,rangeObj,UIAHandler.TextPatternRangeEndpoint_End)<0:
-			tempRange.MoveEndpointByRange(UIAHandler.TextPatternRangeEndpoint_End,rangeObj,UIAHandler.TextPatternRangeEndpoint_End)
-			yield tempRange.clone()
-
-	allowGetFormatFieldsAndTextOnDegenerateUIARanges=False #: _getFormatFieldsAndText should not return anything for degenerate ranges.
-	def _getFormatFieldsAndText(self,tempRange,formatConfig):
-		if not self.allowGetFormatFieldsAndTextOnDegenerateUIARanges and tempRange.compareEndpoints(UIAHandler.TextPatternRangeEndpoint_Start,tempRange,UIAHandler.TextPatternRangeEndpoint_End)==0:
-			return
-		formatField=self._getFormatFieldAtRange(tempRange,formatConfig)
-		if formatConfig["reportSpellingErrors"]:
+	def _getTextFromUIARange(self,range):
+		"""
+		Fetches plain text from the given UI Automation text range.
+		Just calls getText(-1). This only exists to be overridden for filtering.
+		"""
+		return range.getText(-1)
+
+	def _getTextWithFields_text(self,textRange,formatConfig,UIAFormatUnits=UIAFormatUnits):
+		"""
+		Yields format fields and text for the given UI Automation text range, split up by the first available UI Automation text unit that does not result in mixed attribute values.
+		@param textRange: the UI Automation text range to walk.
+		@type textRange: L{UIAHandler.IUIAutomationTextRange}
+		@param formatConfig: the types of formatting requested.
+		@ type formatConfig: a dictionary of NVDA document formatting configuration keys with values set to true for those types that should be fetched.
+		@param UIAFormatUnits: the UI Automation text units (in order of resolution) that should be used to split the text such that formatting won't have mixed values.
+		@type List of UI Automation Text Units 
+		@rtype: a Generator yielding L{textInfos.FieldCommand} objects containing L{textInfos.FormatField} objects, and text strings.
+		"""
+		log.debug("_getTextWithFields_text start")
+		log.debug("Walking by UIA unit %s"%UIAFormatUnits[0])
+		for tempRange in iterUIARangeByUnit(textRange,UIAFormatUnits[0]):
+			text=self._getTextFromUIARange(tempRange)
+			if text:
+				log.debug("Chunk has text. Fetching formatting")
+				try:
+					field=self._getFormatFieldAtRange(tempRange,formatConfig,ignoreMixedValues=len(UIAFormatUnits)<2)

```

Comment:
I suggest changing `< 2` to `== 1`. This makes it clearer that we ignore mixed values if there's only one unit. I don't think it's possible to have 0 units anyway, as the earlier `[0]` would have thrown an exception.